crates/sui-types/src/message_envelope.rs

```diff
@@ -148,16 +149,23 @@ impl<T: Message> Envelope<T, EmptySignInfo> {
         }
     }
 
-    pub fn verify_signature(&self) -> SuiResult {
-        self.data.verify(None)
+    pub fn verify_signature(&self, verify_params: &VerifyParams) -> SuiResult {
+        self.data.verify(verify_params)
     }
 
-    pub fn verify(self) -> SuiResult<VerifiedEnvelope<T, EmptySignInfo>> {
-        self.verify_signature()?;
+    pub fn verify_with_params(
+        self,
+        verify_params: &VerifyParams,
+    ) -> SuiResult<VerifiedEnvelope<T, EmptySignInfo>> {
+        self.verify_signature(verify_params)?;
         Ok(VerifiedEnvelope::<T, EmptySignInfo>::new_from_verified(
             self,
         ))
     }
+
+    pub fn verify(self) -> SuiResult<VerifiedEnvelope<T, EmptySignInfo>> {
```

Comment:
When should this be used versus the one with verify params?

Reply:
I think use this when not the normal authenticator and the above when using zklogin. Agreed the naming is a bit puzzling.

Reply:
I cleaned all of this up to make it much clearer and less error prone.